File: library/src/main/java/com/getstream/sdk/chat/navigation/destinations/ChatDestination.java
package com.getstream.sdk.chat.navigation.destinations;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;

abstract public class ChatDestination {

    protected Context context;

    public ChatDestination(Context context) {
        this.context = context;
    }

    public abstract void navigate();

    protected void start(Intent intent) {
        context.startActivity(intent);
    }

    protected void startForResult(Intent intent, int requestCode) {
        ((Activity) context).startActivityForResult(intent, requestCode);
    }
}

File: library/src/main/java/com/getstream/sdk/chat/navigation/destinations/ChatDestination.kt
package com.getstream.sdk.chat.navigation.destinations

import android.app.Activity
import android.content.Context
import android.content.Intent

public abstract class ChatDestination(protected val context: Context) {
    public abstract fun navigate()

    protected fun start(intent: Intent) {
        context.startActivity(intent)
    }

    protected fun startForResult(intent: Intent, requestCode: Int) {
        check(context is Activity) {
            "startForResult can only be used if your destination uses an Activity as its Context"
        }
        context.startActivityForResult(intent, requestCode)
    }
}
